src/request/export/ExportWords.js

```diff
@@ -112,7 +113,23 @@ class ExportWords extends Nimiq.Observable {
             const passwordBuffer = password ? Utf8Tools.stringToUtf8ByteArray(password) : undefined;
 
             try {
-                key = await KeyStore.instance.get(this._request.keyInfo.id, passwordBuffer);
+                /**
+                 * Because Typescript does not detect the ExportRequest to be different from
+                 * a SimpleRequest (because both extra properties of the ExportRequest are
+                 * optional), we cannot add a new condition for the ExportRequest to the
+                 * Parsed<> transform in types/Keyguard.d.ts.
+                 *
+                 * Because the keyInfo on the parsed request also does not contain any indication
+                 * whether the key is a legacy account or a regular key, we have otherwise no
+                 * idea which IndexedDB store to use.
+                 */
```

Comment:
Is this comment still accurate? The `useLegacyStore` flag seems to always be set to either true or false and is thus not optional.

Reply:
The problem is that the `useLegacyStore` flag only exists on the _parsed_ version of the request, not the incoming external request.

But because the `Parsed<>` transformer does not recognize the ExportRequest as another type other than SimpleRequest, I cannot add a special handling for it, which would not also apply to all other SimpleRequests.

Reply:
If you have a look at https://github.com/nimiq/keyguard-next/blob/668e6944f1ada756b8f9d37d1f9a6069f1aa0351/src/request/remove-key/RemoveKey.js#L28 you'll notice we use it as the same type  indeed. The additional parameter would have to be parsed there as well. Maybe also at other places.
(Side note: would be cleaner to explicitly create a new request object there with an imported constructor, and probably to include a `kind` field in each request.)
My proposal for a quick fix without `ts-ignore`: https://github.com/nimiq/keyguard-next/tree/max/useLegacyStore-type-fix

Reply:
I now resolved it by adding a `KeyInfo.useLegacyStore` flag, which is `false` by default and can be set/get by setters/getters. This way there is no type trickery or type inconsistency required and the only request using it is the export(words).